meta-dts-distro/recipes-dts/dts/dasharo-deploy/dasharo-deploy

```diff
@@ -617,6 +574,128 @@ install() {
+
+    local ver1="$1"
+    local ver2="$2"
+
+    IFS='.' read -r -a arr_ver1 <<< "$ver1"
+    IFS='.' read -r -a arr_ver2 <<< "$ver2"
+
+    if [ ${arr_ver2[0]} -lt ${arr_ver1[0]} ]; then
+        return 0
+    fi
+
+    if [ ${arr_ver2[0]} -gt ${arr_ver1[0]} ]; then
+        return 1
+    fi
+
+    if [ ${arr_ver2[0]} -eq ${arr_ver1[0]} ]; then
+        if [ ${arr_ver2[1]} -lt ${arr_ver1[1]} ]; then
+            return 0
+        fi
+
+        if [ ${arr_ver2[1]} -gt ${arr_ver1[1]} ]; then
+            return 1
+        fi
+
+        if [ ${arr_ver2[1]} -eq ${arr_ver1[1]} ]; then
+            if [ ${arr_ver2[2]} -lt ${arr_ver1[2]} ]; then
+                return 0
+            fi
+
+            if [ ${arr_ver2[2]} -gt ${arr_ver1[2]} ]; then
+              return 1
+            fi
+        fi
+    fi
+}
+
+update() {
+  echo "Checking for the latest Dasharo update available..."
+  echo "Current Dasharo version: $DASHARO_VERSION"
+  echo "Latest available Dasharo version: $DASHARO_REL_VER"
+  compare_versions $DASHARO_VERSION $DASHARO_REL_VER
+  if [ $? -ne 1 ]; then
+    error_exit "No update available for your machine"
+  fi
+
+  BIOS_HASH=$BIOS_HASH_PROD
+  BIOS_LINK=$BIOS_LINK_PROD
+  if [ "$HAVE_EC" = "true" ]; then
+    EC_HASH=$EC_HASH_PROD
+    EC_LINK=$EC_LINK_PROD
+  fi
+
+  display_flashing_warning
+
+  echo "Downloading Dasharo firmware update..."
+  curl --progress-bar -f "$BIOS_LINK" -o $BIOS_UPDATE_FILE
+  error_check "Cannot access $FW_STORE_URL. Please check your internet connection"
+  echo "Checking sha256sum..."
+  sha256sum --check <(echo "$BIOS_HASH")
+  error_check "Failed to verify Dasharo firmware"
+
+  if [ "$HAVE_EC" = "true" ]; then
+    system76_ectool info >> $ERR_LOG_FILE 2>&1
+    if [ $? -ne 0 ]; then
+      error_exit "Device does not have Dasharo EC firmware - cannot continue update!"
+    fi
+  fi
+
+  echo "Updating Dasharo firmware..."
+  print_warning "This will take around 3 minutes. Please be patient and do not reset your computer, or touch keyboard!"
+  flashrom -p "$PROGRAMMER_BIOS" ${FLASH_CHIP_SELECT} ${FLASHROM_ADD_OPT_UPDATE} -w "$BIOS_UPDATE_FILE" &>> $ERR_LOG_FILE
+  error_check "Failed to update Dasharo firmware"
+
+  if [ "$HAVE_EC" = "true" ]; then
+    echo "Dasharo EC update process will start in a moment."
+    sleep 3
+    echo "Your machine will shut down automatically once it finishes."
+    sleep 3
+    print_warning "This will take around 1 minute. Please be patient and do not reset your computer, or touch keyboard!"
+    echo "Updating Dasharo EC firmware..."
+    update_ec
+  fi
+
+  print_green "Successfully updated Dasharo firmware"
+
+  echo -n "Syncing disks... "
+  sync
+  echo "Done."
+  echo "The computer will reboot automatically in 5 seconds"
+  sleep 0.5
+  echo "Rebooting in 5s:"
+  echo "5..."
+  sleep 1
+  echo "4..."
+  sleep 1
+  echo "3..."
+  sleep 1
+  echo "2..."
+  sleep 1
+  echo "1..."
+  sleep 0.5
+  echo "Rebooting"
+  sleep 1
+  ${CMD_REBOOT}
```

Comment:
@macpijan reboot is enough even if we flash ec?

Reply:
We will never reach this code when we update EC. system76_tool flash automatically shuts down after flashing was done.